apps/condo/domains/billing/schema/ValidateQRCodeService.js

```diff
@@ -65,43 +87,50 @@ const ValidateQRCodeService = new GQLCustomSchema('ValidateQRCodeService', {
         },
         {
             access: true,
-            type: 'type ValidateQRCodeOutput { qrCodeFields: JSON! }',
+            type: 'type ValidateQRCodeLastReceiptDataOutput { id: ID!, period: String!, toPay: String! }',
+        },
+        {
+            access: true,
+            type: 'type ValidateQRCodeFeesOutput { explicitServiceCharge: String, explicitFee: String }',
+        },
+        {
+            access: true,
+            type: 'type ValidateQRCodeOutput { qrCodeFields: JSON!, lastReceiptData: ValidateQRCodeLastReceiptDataOutput, explicitFees: ValidateQRCodeFeesOutput!, amount: String! }',
         },
     ],
 
     mutations: [
         {
             access: access.canValidateQRCode,
             schema: 'validateQRCode(data: ValidateQRCodeInput!): ValidateQRCodeOutput',
-            resolver: async (parent, /** @type {ValidateQRCodeInputArgs} */ args, context) => {
+            resolver: async (parent, /** ValidateQRCodeInputArgs */ args, context) => {
                 const { data } = args
                 const { qrCode } = data
 
-                const ip = context.req.ip
-                await redisGuard.checkCustomLimitCounters(
-                    `validate-QR-code-${ip}`,
-                    BILLING_VALIDATE_QR_CODE_WINDOW,
-                    MAX_CLIENT_VALIDATE_QR_CODE_BY_WINDOW,
-                )
-
-                const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(qrCode)
-                const requisitesStr = get(matches, ['groups', 'requisitesStr'], '')
-
-                // TODO(AleX83Xpert): maybe decode requisitesStr
-                // Need to test the result of scanning from mobile devices
-                // https://encoding.spec.whatwg.org/#koi8-r
-                // https://encoding.spec.whatwg.org/#windows-1251
-                // const encodingTag = get(matches, ['groups', 'encodingTag'])
-                // const encoding = get(['windows-1251', 'utf-8', 'koi8-r'], encodingTag, 'utf-8')
+                const currentUserId = get(context, ['authedItem', 'id'])
+                if (!currentUserId) {
+                    const ip = context.req.ip
+                    await redisGuard.checkCustomLimitCounters(
```

Comment:
Why are you skipping this check for an authorized user?
They can spam too :)

Reply:
If so, we can find these users and maybe spread limit for registered users too.